tests/test_measure.py

```diff
@@ -304,60 +280,6 @@ def circuit():
         assert result[2].dtype == np.dtype("float")
         assert np.array_equal(result[2].shape, (n_sample,))
 
-    def test_sample_default_n(self, tol):
-        """Test the return type and shape of sampling multiple works
-           in combination with expvals and vars"""
-
-        n_shots = 10
-        dev = qml.device("default.qubit", wires=1, shots=n_shots)
-
-        @qml.qnode(dev)
-        def circuit():
-            return qml.sample(qml.PauliZ(0))
-
-        result = circuit()
-
-        # If all the dimensions are equal the result will end up to be a proper rectangular array
-        assert np.array_equal(result.shape, (n_shots,))
-
-    def test_sample_exception_device_context_missing(self):
```

Comment:
How come this test was removed? I expected it to show up in `test_device.py` now, but didn't see it there

Reply:
Because it tested for an exception that was removed.

Reply:
This particular edge case (trying to sample from a QNode that doesn't have a device attached to it) seems to be independent of the actual PR scope, so I was surprised it was removed (I assumed it was maybe just removed from this file and placed somewhere more appropriate).

If we were checking for this exception before, why stop now? If it is a legitimate edge case that might occur to a user, isn't it better to have a descriptive exception message?

Reply:
This particular edge case was there even before sampling was introduced (think of taking an expval of a QNode that has no device attached). The measure routines will yield no problem, because accessing the underlying device is not necessary at this stage. It only caused a problem in `sample` because we pulled the default number of samples from the underlying device (which is not the case anymore, because the number of samples is now a device internal).

I think testing for the current context to have a device attached to it is unnecessary. If we would want to do so, it would have to be in `QNode.__init__` where the device is passed to the `QNode`. We did not need that up to now and I don't think it is important for the future.